### ISSUE TITLE
detechtig potential slash 

### ISSUE THREAD

**Issue by 2tata** [koalaman/shellcheck#808]
#### For new checks and feature suggestions

- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this

Local version:
~ % shellcheck --version
ShellCheck - shell script analysis tool
version: 0.4.5
license: GNU General Public License, version 3
website: http://www.shellcheck.net

and I tried on  shellcheck.net  as well

- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Not really. One issue can maybe involted #271 

Yes I tried on  shellcheck.net  as well

Maybe shellcheck can warn for potential double slashes like here:

Problematic code
```sh
# Set directorie from argument, or use default.
targetdir=${1-.}

 # Check for rejects...
 if [ "$(find "$targetdir/" '(' -name '*.rej' -o -name '.*.rej' ')' -print)" ] ; then
     echo "Aborting.  Reject files found."
     exit 1
 fi
 
 # Remove backup files
 find "$targetdir/" '(' -name '*.orig' -o -name '.*.orig' ')' -exec rm -f {} \;

```
correct code:
```sh
# Set directorie from argument, or use default.
targetdir=${1-.}

 # Check for rejects...
 if [ "$(find "${targetdir%/}/" '(' -name '*.rej' -o -name '.*.rej' ')' -print)" ] ; then
     echo "Aborting.  Reject files found."
     exit 1
 fi
 
 # Remove backup files
 find "${targetdir%/}/" '(' -name '*.orig' -o -name '.*.orig' ')' -exec rm -f {} \;

```
cheers
Tarek

**Comment by Herst** [comment]
AFAIK, there is no harm in multiple successive slashes (as long as a path does not begin with them) because as per standard they should be interpreted just like single one at that position.

**Comment by 2tata** [comment]
Ah thanks, I ware not informed about, that double slashes will always interpreted as ones.
butt in case is Targetdir a list of URLs ? so double slash can cause the functionallity

**Comment by kurahaupo** [comment]
The "find" command does not need to have trailing slashes on filenames at
all.

Aggregating the entire output of a command before checking "is it
(non-)empty" [ -n "$(...)" ] is a rather "heavy" approach. Instead, stop as
soon as you know that you've found at least one item.

Find is not likely to produce URLs, but even there it does not matter if
you have multiple slashes, except immediately after the scheme tag.

On 26 Dec. 2016 02:35, "Tata" <notifications@github.com> wrote:

Ah thanks, I ware not informed about, that double slashes will always
interpreted as ones.
butt in case is Targetdir a list of URLs ? so double slash can cause the
functionallity

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/koalaman/shellcheck/issues/808#issuecomment-269127411>,
or mute the thread

**Comment by koalaman** [comment]
You don't worry about multiple slashes, they're guaranteed to work the same as a single slash on a local fs. Most webservers by extension work the same way, such as  https://github.com//koalaman////shellcheck//issues////808
